### ISSUE TITLE
Issues with Linux integration

### ISSUE THREAD

**Issue by dmarcuse** [DoubleDoorDevelopment/GlobalSettings#3]
Hey, I'm really glad to see that you're working on improved Linux support, but it seems like there's a couple issues right now:

1. The way you check for Linux is incorrect - you're currently checking the `os.arch` property instead of the `os.name` property. Here's an example:

```java
System.getProperty("os.name").toLowerCase().contains("linux")
```

2. The `XDG_CONFIG_HOME` environment variable may not always be set, resulting in potentially unintended behavior [here](https://github.com/DoubleDoorDevelopment/GlobalSettings/blob/master/src/main/java/net/doubledoordev/globalsettings/Utils.java#L50) when `null` is returned from `System.getenv`. According to the [specification](https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html), `$HOME/.config` should be used in this case - e.g., the following code to get the parent folder as a `File`:

```java
new File(Optional.ofNullable(System.getenv("XDG_CONFIG_DIR")).orElseGet(() -> System.getProperty("user.home") + File.separator + ".config"))
```

Alternatively, there is the [appdirs](https://github.com/harawata/appdirs) library which handles all of this for you, as well as supporting OSX and Windows better, although it does seem rather overkill for something this simple considering it depends on JNA.

Thanks again for your work on this!

**Comment by Claycorp** [comment]
Dammit! I had a feeling I was doing it all wrong... (I should have waited a day to have someone that has a Linux setup test it.)

I will get this fixed ASAP. Thanks for hanging with me on this, I'm no pro at java or forge as you can likely tell from looking at the source.

I'll stick with treating Linux specially without anything extra. IMO I could have just went with the JVM option as that would have been sufficient but it exits now so I might as well keep it!

**Comment by dmarcuse** [comment]
All good, I'm glad to help :)

**Comment by Claycorp** [comment]
Should be fixed with 015892282e75d68ba6a3251994e200214c260d61
